lib/cdo/shared_constants.rb

```diff
@@ -0,0 +1,16 @@
+# A set of constants shared between ruby and JS via /shared/config/sharedConstants.json
+require 'json'
+
+module SharedConstants
+  @@json = nil
+
+  def self.json
+    return @@json if @@json
+
+    file = File.read(File.join(__dir__, '../../shared/config/constants.json'))
+    # @@json = JSON.parse(file)
+    @@json = JSON.parse(file, object_class: OpenStruct)
```

Comment:
I'd favor a build step that creates a static rb file _and_ a static js file (like our color.js file) if only because it will improve editor support for these constants.  One of the things I dislike about current JavaScript enum utility is that my editor doesn't recognize any of the enum values.